chain/chain/src/garbage_collection.rs

```diff
@@ -216,10 +216,6 @@ impl ChainStore {
                         *block_hash,
                     )?;
                     gc_blocks_remaining -= 1;
-                } else {
-                    return Err(Error::GCError(
-                        "block on canonical chain shouldn't have refcount 0".into(),
-                    ));
```

Comment:
How did it work previously though, given that we always store range of blocks and for the first of them refcount is 0?
Can we make state sync impl aligned with current logic, maybe add `inc_block_refcount` to some other place?